src/wait_group.cr

```diff
@@ -59,16 +59,26 @@ class WaitGroup
   # decrement the counter, to make sure that the counter may never inadvertently
   # reach zero before all fibers are done.
   def add(n : Int32 = 1) : Nil
-    new_value = @counter.add(n) + n
-    return if new_value > 0
+    counter = @counter.get(:acquire)
+    new_counter = uninitialized Int32
```

Comment:
**nitpick:** It might be better to initialize to `nil` because then the compiler will complain about a logic error that could leave this variable being unassigned a real value.

Reply:
I'm not fond of introducing nilables just for the sake of it (we immediately call `#not_nil!`).

I tried alternatives, like `val = loop { val = x; break val }` but "can't use variable inside assignment" (despite the value being defined in the block, so it could be understood as shadowing the variable). I didn't want to use a `tmp` or `val2` variable name.

So I moved `new_counter` after the loop, and do the `+` twice :shrug: 

I still believe `uninitialized` was fine. It was merely out constraining the variable scope and it was never read before it was defined. In fact, a linter could prove it.

Reply:
Actually, there should be no need for `not_nil!`. The compiler is capable of figuring out that the loop runs at least once and it assigns to `new_counter`, so it's typed without `Nil` afterwards.
So literally just changing `uninitialized Int32` to `nil` should've worked. I'm fine with the refactored version, though.